tests/test_git_fetch.rs

```diff
@@ -0,0 +1,368 @@
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    @ 230dd059e1b0 (no description set) 
+    o 000000000000 (no description set) 
+    "###);
+    insta::assert_snapshot!(test_env.jj_cmd_success(&target_jj_repo_path, &["git", "fetch"]), @"");
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o 43fe77b5815e descr_for_b b
+    | o cb1fe5d97488 descr_for_a2 a2
+    |/  
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    o 9929b494c411 trunk1 master trunk1
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+
+    // Change the target repo
+    let source_log = create_trunk2_and_rebase_branches(&test_env, &source_git_repo_path);
+    insta::assert_snapshot!(source_log, @r###"
+    Source git repo contents:
+    o 90ded7086076 descr_for_b b
+    | o d81785eb05b3 descr_for_a2 a2
+    |/  
+    | o 4b6d31d2dbab descr_for_a1 a1
+    |/  
+    @ baf8ee894f05 descr_for_trunk2 trunk2
+    o 9929b494c411 trunk1 master trunk1
+    o 000000000000 (no description set) 
+    "###);
+
+    // Our repo before and after fetch
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o 43fe77b5815e descr_for_b b
+    | o cb1fe5d97488 descr_for_a2 a2
+    |/  
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    o 9929b494c411 trunk1 master trunk1
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+    insta::assert_snapshot!(test_env.jj_cmd_success(&target_jj_repo_path, &["git", "fetch"]), @"");
+    insta::assert_snapshot!(get_log_output(&test_env, &target_jj_repo_path), @r###"
+    o 90ded7086076 descr_for_b b
+    | o d81785eb05b3 descr_for_a2 a2
+    |/  
+    | o 4b6d31d2dbab descr_for_a1 a1
+    |/  
+    o baf8ee894f05 descr_for_trunk2 trunk2
+    o 9929b494c411 trunk1 master trunk1
+    | @ 230dd059e1b0 (no description set) 
+    |/  
+    o 000000000000 (no description set) 
+    "###);
+}
+
+#[test]
+fn test_git_fetch_some() {
+    let test_env = TestEnvironment::default();
+    let source_git_repo_path = test_env.env_root().join("source");
+    let _git_repo = git2::Repository::init(source_git_repo_path.clone()).unwrap();
+
+    // Clone an empty repo
+    let stdout =
+        test_env.jj_cmd_success(test_env.env_root(), &["git", "clone", "source", "target"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Fetching into new repo in "$TEST_ENV/target"
+    Nothing changed.
+    "###);
+    let target_jj_repo_path = test_env.env_root().join("target");
+
+    let source_log =
+        create_colocated_repo_and_branches_from_trunk1(&test_env, &source_git_repo_path);
+    insta::assert_snapshot!(source_log, @r###"
+       ===== Source git repo contents =====
+    @ 43fe77b5815e descr_for_b b
+    | o cb1fe5d97488 descr_for_a2 a2
+    |/  
+    | o 5f35ea16f836 descr_for_a1 a1
+    |/  
+    o 9929b494c411 trunk1 master trunk1
```

Comment:
TODO: I can't figure out why the description here is `trunk1` rather than `descr_for_trunk1`. Maybe it'll be clearer in the morning.

Reply:
Fixed.